kotlin-asyncapi-spring-web/src/main/kotlin/org/openfolder/kotlinasyncapi/springweb/context/annotation/processor/MappingUtils.kt

```diff
@@ -0,0 +1,161 @@
+        http = this@toMessageBindings.http.takeUnless { it.default }?.toHTTP()
+        kafka = this@toMessageBindings.kafka.takeUnless { it.default }?.toKafka()
+        anypointmq = this@toMessageBindings.anypointmq.takeUnless { it.default }?.toAnypointMQ()
+        amqp = this@toMessageBindings.amqp.takeUnless { it.default }?.toAMQP()
+        mqtt = this@toMessageBindings.mqtt.takeUnless { it.default }?.toMQTT()
+        ibmmq = this@toMessageBindings.ibmmq.takeUnless { it.default }?.toIBMMQ()
+    }
+
+internal fun MessageTrait.toMessageTrait(): org.openfolder.kotlinasyncapi.model.channel.MessageTrait =
+    org.openfolder.kotlinasyncapi.model.channel.MessageTrait().apply {
+        messageId = this@toMessageTrait.messageId.takeIf { it.isNotEmpty() }
+        schemaFormat = this@toMessageTrait.schemaFormat.takeIf { it.isNotEmpty() }
+        contentType = this@toMessageTrait.contentType.takeIf { it.isNotEmpty() }
+        name = this@toMessageTrait.name.takeIf { it.isNotEmpty() }
+        title = this@toMessageTrait.title.takeIf { it.isNotEmpty() }
+        summary = this@toMessageTrait.summary.takeIf { it.isNotEmpty() }
+        description = this@toMessageTrait.description.takeIf { it.isNotEmpty() }
+        correlationId = this@toMessageTrait.correlationId.takeUnless { it.default }?.toCorrelationID()
+        tags = this@toMessageTrait.tags.takeUnless { it.isEmpty() }?.toTagsList()
+        externalDocs = this@toMessageTrait.externalDocs.takeUnless { it.default }?.toExternalDocumentation()
+        examples = this@toMessageTrait.examples.takeUnless { it.isEmpty() }?.toMessageExamplesList()
+        bindings = this@toMessageTrait.bindings.takeUnless { it.default }?.toMessageBindings()
+        headers = this@toMessageTrait.headers.takeUnless { it.default }?.implementation?.let {
+            Reference().apply { ref("#/components/schemas/${it.simpleName}") }
+        }
+    }
+
+internal fun MessageBinding.HTTP.toHTTP(): org.openfolder.kotlinasyncapi.model.channel.MessageBinding.HTTP =
+    org.openfolder.kotlinasyncapi.model.channel.MessageBinding.HTTP().apply {
+        bindingVersion = this@toHTTP.bindingVersion.takeIf { it.isNotEmpty() }
+        headers = this@toHTTP.headers.takeUnless { it.default }?.implementation?.let {
+            Reference().apply { ref("#/components/schemas/${it.simpleName}") }
+        }
+    }
+
+internal fun MessageBinding.Kafka.toKafka(): org.openfolder.kotlinasyncapi.model.channel.MessageBinding.Kafka =
+    org.openfolder.kotlinasyncapi.model.channel.MessageBinding.Kafka().apply {
+        bindingVersion = this@toKafka.bindingVersion.takeIf { it.isNotEmpty() }
+        key = this@toKafka.key.takeUnless { it.default }?.implementation?.let {
+            Reference().apply { ref("#/components/schemas/${it.simpleName}") }
+        }
+    }
+
+internal fun MessageBinding.AnypointMQ.toAnypointMQ(): org.openfolder.kotlinasyncapi.model.channel.MessageBinding.AnypointMQ =
```

Comment:
Where does SQS/SNS fit in here? Is it this Anypoint?

Reply:
sqs and sns also have message bindings but they are not implemented by AsyncAPI for now https://github.com/asyncapi/bindings/blob/master/sqs/README.md. that is why we dont have a mapping here.